src/algorithm/translate.rs

```diff
@@ -0,0 +1,196 @@
+use num_traits::{Float, FromPrimitive};
+use types::{Point, Polygon, LineString, Line, MultiPoint, MultiPolygon, MultiLineString};
+
+// translate a slice of points by the given offsets
+fn translation_matrix<T>(x_offset: T, y_offset: T, points: &[Point<T>]) -> Vec<Point<T>>
+    where T: Float
+{
+    points
+        .iter()
+        .map(|point| {
+                 let new_x = point.x() + x_offset;
+                 let new_y = point.y() + y_offset;
+                 Point::new(new_x, new_y)
```

Comment:
My WIP is here: https://github.com/rory/rust-geo/tree/coord_change. Specifically you can see the file here https://github.com/rory/rust-geo/commit/b7a01f5a79eca2de7b581aa192b023778b122787 . My idea is to allow you to pass in a function which will take a `Coordinate` and return a new `Coordinate`, which gives you maximum flexibility. I don't like the name (`coord_change`), but can't think of much better.